src/debug_utils.cc

```diff
@@ -0,0 +1,226 @@
+#include "debug_utils.h"
+#include "node_internals.h"
+
+#ifdef __POSIX__
+#if defined(__linux__)
+#include <features.h>
+#endif
+
+#if defined(__linux__) && !defined(__GLIBC__) || \
+    defined(__UCLIBC__) || \
+    defined(_AIX)
+#define HAVE_EXECINFO_H 0
+#else
+#define HAVE_EXECINFO_H 1
+#endif
+
+#if HAVE_EXECINFO_H
+#include <cxxabi.h>
+#include <dlfcn.h>
+#include <execinfo.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <stdio.h>
+#endif
+
+#else  // __POSIX__
+
+#include <windows.h>
+#include <dbghelp.h>
+
+#endif  // __POSIX__
+
+namespace node {
+
+#ifdef __POSIX__
+#if HAVE_EXECINFO_H
+class PosixSymbolDebuggingContext final : public NativeSymbolDebuggingContext {
+ public:
+  PosixSymbolDebuggingContext() : pagesize_(getpagesize()) { }
+
+  SymbolInfo LookupSymbol(void* address) override {
+    Dl_info info;
+    const bool have_info = dladdr(address, &info);
+    SymbolInfo ret;
+    if (!have_info)
+      return ret;
+
+    if (info.dli_sname != nullptr) {
+      if (char* demangled = abi::__cxa_demangle(info.dli_sname, 0, 0, 0)) {
+        ret.name = demangled;
+        free(demangled);
+      } else {
+        ret.name = info.dli_sname;
+      }
+    }
+
+    if (info.dli_fname != nullptr) {
+      ret.filename = info.dli_fname;
+    }
+
+    return ret;
+  }
+
+  bool IsMapped(void* address) override {
+    void* page_aligned = reinterpret_cast<void*>(
+        // NOLINTNEXTLINE(whitespace/operators)
+        reinterpret_cast<uintptr_t>(address) &~ (pagesize_ - 1));
+    return msync(page_aligned, pagesize_, 0) == 0;
```

Comment:
I'm surprised this works; passing in no flags is technically an error (they're non-zero on linux and macos.)

Reply:
@bnoordhuis I guess this could be made `MS_ASYNC`? Reading the Linux man page, it seems like that’s how Linux interprets `0` anyway … 

I’m switching and seeing how that fares in CI.